### DISCUSSION TITLE
MAUI needs a good (selfhosted) serverless partner.

### DISCUSSION THREAD

**Discussion by Qws** [dotnet/maui#22772] [Ideas]
### Description

I don't want to build backend service from scratch. So I've been spending some time finding serverless backend solution for my .NET MAUI app.

Firebase, Supabase, Pocketbase... they all have official Flutter SDK support but not MAUI SDK support. This makes MAUI a bit less attractive to use. It even makes me want to move towards Google's Flutter, despite really loving XAML and MVVM.

I'm not asking for Microsoft to build a Firebase for MAUI. I currently don't want to deal with Firebase or Azure.

Pocketbase is fully open source, it works super nice because you can **self host it** with single executable. So you are not stuck behind Google's or Supabase servers.

Having serverless backend that you can selfhost without unpredictable cost is great for small developers.

I know Microsoft invest money in open source projects to push their own tech (Godot Game Engine -> C# language support)...

Therefore I would like to see Microsoft invest in a MAUI SDK for Pocketbase. So that people can self-host their Serverless solution and use MAUI as their frontend.

### Public API Changes

Not a frontend API change.

MAUI needs an easy to setup backend solution.

I wish Microsoft could help realize a .NET SDK for Pocketbase, so that MAUI developers don't need to worry about expensive or technically challenging backend solutions.

### Intended Use-Case

Building your own backend is frustrating and challenging. Security, email service, database, authentication.

We need simple to use, serverless backend solutions for our MAUI application.

As I said before, a Flutter app can use Google's official Firebase SDK (for user authentication, database, storage, functions etc). https://docs.flutter.dev/development/data-and-backend/firebase they even promote it in their docs..

If Flutter dev doesn't want to be locked behind Google's or Supabase servers, they can simply download Pocketbase and selfhost it.

MAUI is sadly weak with backend support. We need a straightforward, easy to use backend solutions for MAUI.

**Comment by ghost** [comment]
We've moved this issue to the Backlog milestone. This means that it is not going to be worked on for the coming release. We will reassess the backlog following the current release and consider this item at that time. To learn more about our issue management process and to have better expectation regarding different types of issues you can read our [Triage Process](https://github.com/dotnet/maui/blob/main/docs/TriageProcess.md).

**Comment by rpconer** [comment]
very right i was looking for the same thing. I am experimenting if aws + MAUI can work together

**Comment by HobDev** [comment]
Supabase supports Maui. This [guide](https://hackernoon.com/creating-a-books-tracker-app-with-net-maui-and-supabase) teach how to create Mauii app with Supabase backend.
Mongo DB also have [.NET SDK](https://www.mongodb.com/docs/realm/sdk/dotnet/) .
